lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -90,4 +137,42 @@ defmodule ExUnit.Server do
   def handle_cast(request, config) do
     super(request, config)
   end
+
+  def handle_info({:DOWN, ref, _, _, _}, config) do
+    refs = Map.delete(config.log_captures, ref)
+    maybe_add_console(refs, config.log_status)
+    remove_capture(ref)
+    {:noreply, %{config | log_captures: refs}}
+  end
+
+  def handle_info({:gen_event_EXIT, _, :normal}, config) do
+    {:noreply, config}
+  end
+
+  def handle_info({:gen_event_EXIT, {Console, ref}, reason}, config) do
+    refs = Map.put(config.log_captures, ref, reason)
+    maybe_add_console(refs, config.log_status)
+    {:noreply, %{config | log_captures: refs}}
+  end
+
+  def handle_info(msg, state) do
+    super(msg, state)
+  end
+
+  defp remove_capture(ref) do
+    case GenEvent.remove_handler(Logger, {Console, ref}, nil) do
+      {:error, :not_found} ->
+        receive do
+          {:gen_event_EXIT, {Console, ^ref}, reason} ->
```

Comment:
I think there is a race condition here where the capture handler crashes and then the capturing process crashes. In this situation the handler is gone when the `:DOWN` is received and this message will never be received. We can fix this by doing an `after 0 -> not_found`.